modules/account-lib/src/mpc/tss.ts

```diff
@@ -0,0 +1,300 @@
+ *    signing protocol is complete.
+ *
+ * ======================
+ * EdDSA Signing
+ * ======================
+ * 1. The parties from key generation decide they want to sign something. They begin the signing protocol
+ *    by generating shares of an ephemeral key.
+ *
+ *    a) Each signer uses his p-share and the j-shares stored for other players to generate his signing share.
+ *    b) This results in each signer having a private x-share and public r-shares.
+ *
+ * 2. Signers distribute their r-shares to other signers.
+ * 3. After exchanging r-shares, each signer signs their share of the ephemeral key using their private
+ *    x-share with the r-shares from other signers.
+ * 4. This results in each signer having a public g-share which they send to the other signers.
+ * 5. After the signers broadcast their g-shares, the final signature can be re-constructed independently.
+ */
+const assert = require('assert');
+import Ed25519Curve from './curves';
+import { sha512 } from 'js-sha512';
+import Shamir from './shamir';
+import { randomBytes as cryptoRandomBytes } from 'crypto';
+
+const convertObjectHexToBuffer = (object: Record<string, any>, keys: string[]) => {
+  const result: any = object;
+  keys.forEach((key) => {
+    result[key] = Buffer.from(object[key], 'hex');
+  });
+  return result;
+};
+
+interface UShare {
+  i: string;
+  y: string;
+  u: string;
+  prefix: string;
+}
+
+interface YShare {
+  i: string;
+  j: string;
+  y: string;
+  u: string;
+}
+
+interface KeyShare {
+  uShare: UShare;
+  yShares: Record<string, YShare>;
+}
+
+interface PShare {
+  i: string;
+  x: string;
+  y: string;
+  prefix: string;
+}
+
+interface JShare {
+  i: string;
+  j: string;
+}
+
+interface KeyCombine {
+  pShare: PShare;
+  jShares: Record<string, JShare>;
+}
+
+interface XShare {
+  i: string;
+  y: string;
+  x: string;
+  r: string;
+  R: string;
+}
+
+interface RShare {
+  i: string;
+  j: string;
+  r: string;
+  R: string;
+}
+
+interface SignShare {
+  xShare: XShare;
+  rShares: Record<string, RShare>;
+}
+
+interface GShare {
+  i: string;
+  y: string;
+  gamma: string;
+  R: string;
+}
+
+interface Signature {
+  y: string;
+  R: string;
+  sigma: string;
+}
+
+const Eddsa = async () => {
+  const ed25519 = await Ed25519Curve();
+  const shamir = Shamir(ed25519);
+
+  const keyShare = (index: number, threshold: number, numShares: number): KeyShare => {
+    assert(index > 0 && index <= numShares);
+    const sk = cryptoRandomBytes(32);
+    const h = Buffer.from(sha512.digest(sk));
+    const zeroBuffer = Buffer.alloc(64 - h.length);
+    const combinedBuffer = Buffer.concat([zeroBuffer, h]);
+
+    let uBuffer = combinedBuffer.slice(0, 32);
+    uBuffer[0] &= 248;
+    uBuffer[31] &= 63;
+    uBuffer[31] |= 64;
+
+    // using big endian
+    const zeroBuffer32 = Buffer.alloc(32);
+    uBuffer = Buffer.concat([zeroBuffer32, uBuffer]);
+    uBuffer.reverse();
```

Comment:
I believe this based on an incorrect implementation of private key generation from ggmpc. It should be:

```
uBuffer = Buffer.concat([uBuffer, zeroBuffer32])
```

Note that we don't reverse the u buffer after concating the zeros *after* the u value.

You can test this against ggmpc 0.0.7 and it should work

Reply:
good catch, it passes tests with either version. updated so that uBuffer and prefix aren't using big endian